app/src/main/java/app/grapheneos/camera/capturer/ImageSaver.kt

```diff
@@ -126,13 +126,12 @@ class ImageSaver(
     private fun saveImage() {
         try {
             saveImageInner()
+            imageCapturer.mActivity.thumbnailLoaderExecutor.executeIfAlive(this::generateThumbnail)
```

Comment:
This is bad code style.
Code that doesn't throw exceptions shouldn't be moved into `try {}` block.
Add a `finally{}' block and move `photosInQueue.getAndDecrement()` there.

I still disagree with your approach of tracking the queue length, but don't wan't to argue about it anymore.

Reply:
> I still disagree with your approach of tracking the queue length, but don't want to argue about it anymore.

The approach you're asking me to take causes the queue to have more images than it otherwise should. I have already explained why that approach is being used already a couple of time earlier. We can ignore it and have some workarounds, but that still leaves the code broken.

Reply:
> I still disagree with your approach of tracking the queue length, but don't want to argue about it anymore.

The approach you're asking me to take causes the queue to have more images than it otherwise should. I have already explained why that approach is being used already a couple of time earlier. We can ignore it and have some workarounds, but that still leaves the code broken.

The finally block has been added to the code. Thanks for your valuable time @muhomorr.